DicomRtImportExport/Logic/vtkSlicerDicomRtReader.cxx

```diff
@@ -558,6 +580,62 @@ void vtkSlicerDicomRtReader::vtkInternal::LoadRTPlan(DcmDataset* dataset)
 
+      // Get multi leaf collimators parameters: distance, number of leaf pairs, leaf position boundaries
+      DRTBeamLimitingDeviceSequenceInRTBeamsModule& rtBeamLimitingDeviceSequence = 
+        currentBeamSequenceItem.getBeamLimitingDeviceSequence();
+      if (rtBeamLimitingDeviceSequence.gotoFirstItem().good())
+      {
+        do
+        {
+          DRTBeamLimitingDeviceSequenceInRTBeamsModule::Item &collimatorItem =
+            rtBeamLimitingDeviceSequence.getCurrentItem();
+          if (collimatorItem.isValid())
+          {
+            OFString deviceType("");
+            Sint32 nofPairs = -1;
+
+            collimatorItem.getRTBeamLimitingDeviceType(deviceType);
+            if (deviceType == "MLCX" || deviceType == "MLCY")
+            {
+              Float64 sourceToMLCDistance = -1.;
+              collimatorItem.getSourceToBeamLimitingDeviceDistance(sourceToMLCDistance);
+              
+              OFCondition getNumberOfPairsCondition = collimatorItem.getNumberOfLeafJawPairs(nofPairs);
+              if (getNumberOfPairsCondition.good())
+              {
+                OFVector<vtkTypeFloat64> leafPositionBoundaries;
+                OFCondition getBoundariesCondition = collimatorItem.getLeafPositionBoundaries(leafPositionBoundaries);
+                if (getBoundariesCondition.good())
+                {
+                  double& distanceX = MultiLeafCollimator[0].Distance;
+                  double& distanceY = MultiLeafCollimator[1].Distance;
+                  unsigned int& pairsX = MultiLeafCollimator[0].NumberOfLeafJawPairs;
+                  unsigned int& pairsY = MultiLeafCollimator[1].NumberOfLeafJawPairs;
+                  std::vector<double>& boundX = MultiLeafCollimator[0].LeafPositionBoundary;
+                  std::vector<double>& boundY = MultiLeafCollimator[1].LeafPositionBoundary;
+                  if (deviceType == "MLCX")
+                  {
+                    distanceX = sourceToMLCDistance;
+                    pairsX = nofPairs;
+                    boundX.resize(leafPositionBoundaries.size());
+                    std::copy( leafPositionBoundaries.begin(), leafPositionBoundaries.end(), boundX.begin());
+                    vtkDebugWithObjectMacro( this->External, "LoadRTPlan: MLCX number of leaf/jaw pairs: " << nofPairs);
+                  }
+                  else if (deviceType == "MLCY")
+                  {
+                    distanceY = sourceToMLCDistance;
+                    pairsY = nofPairs;
+                    boundY.resize(leafPositionBoundaries.size());
+                    std::copy( leafPositionBoundaries.begin(), leafPositionBoundaries.end(), boundY.begin());
+                    vtkDebugWithObjectMacro( this->External, "LoadRTPlan: MLCY number of leaf/jaw pairs: " << nofPairs);
+                  }
+                }
+              }
+            }
+          }
+        } while (rtBeamLimitingDeviceSequence.gotoNextItem().good());
+      }

```

Comment:
Very nice code btw, thanks for paying close attention to style!

Reply:
I will fix that.